File: app/src/main/java/com/brainwellnessspa/dashboardModule/models/ViewAllAudioListModel.java
package com.brainwellnessspa.dashboardModule.models;

import android.os.Parcel;
import android.os.Parcelable;

import com.google.gson.annotations.Expose;
import com.google.gson.annotations.SerializedName;

import java.util.ArrayList;

public class ViewAllAudioListModel {
    @SerializedName("ResponseData")
    @Expose
    private ResponseData responseData;
    @SerializedName("ResponseCode")
    @Expose
    private String responseCode;
    @SerializedName("ResponseMessage")
    @Expose
    private String responseMessage;
    @SerializedName("ResponseStatus")
    @Expose
    private String responseStatus;

    public ResponseData getResponseData() {
        return responseData;
    }

    public void setResponseData(ResponseData responseData) {
        this.responseData = responseData;
    }

    public String getResponseCode() {
        return responseCode;
    }

    public void setResponseCode(String responseCode) {
        this.responseCode = responseCode;
    }

    public String getResponseMessage() {
        return responseMessage;
    }

    public void setResponseMessage(String responseMessage) {
        this.responseMessage = responseMessage;
    }

    public String getResponseStatus() {
        return responseStatus;
    }

    public void setResponseStatus(String responseStatus) {
        this.responseStatus = responseStatus;
    }

    public static class ResponseData  implements Parcelable {


        @SerializedName("HomeAudioID")
        @Expose
        public String homeAudioID;
        @SerializedName("View")
        @Expose
        public String view;
        @SerializedName("UserId")
        @Expose
        public String userId;
        @SerializedName("Details")
        @Expose
        public ArrayList<Detail> details = null;

        public String getHomeAudioID() {
            return homeAudioID;
        }

        public void setHomeAudioID(String homeAudioID) {
            this.homeAudioID = homeAudioID;
        }

        public String getView() {
            return view;
        }

        public void setView(String view) {
            this.view = view;
        }

        public String getCoUserId() {
            return userId;
        }

        public void setCoUserId(String userId) {
            this.userId = userId;
        }

        public ArrayList<Detail> getDetails() {
            return details;
        }

        public void setDetails(ArrayList<Detail> details) {
            this.details = details;
        }

        public static Creator<MainAudioModel.ResponseData> getCREATOR() {
            return CREATOR;
        }

        public static final Creator<MainAudioModel.ResponseData> CREATOR = new Creator<MainAudioModel.ResponseData>() {
            @Override
            public MainAudioModel.ResponseData createFromParcel(Parcel in) {
                return new MainAudioModel.ResponseData(in);
            }

            @Override
            public MainAudioModel.ResponseData[] newArray(int size) {
                return new MainAudioModel.ResponseData[size];
            }
        };

        @Override
        public int describeContents() {
            return 0;
        }

        @Override
        public void writeToParcel(Parcel parcel, int i) {
            parcel.writeString(homeAudioID);
            parcel.writeString(view);
            parcel.writeString(userId);
        }

        public static class Detail  implements Parcelable {
            @SerializedName("ID")
            @Expose
            public String ID;
            @SerializedName("Name")
            @Expose
            public String name;
            @SerializedName("AudioFile")
            @Expose
            public String audioFile;
            @SerializedName("ImageFile")
            @Expose
            public String imageFile;
            @SerializedName("AudioDuration")
            @Expose
            public String audioDuration;
            @SerializedName("AudioDirection")
            @Expose
            public String audioDirection;
            @SerializedName("Audiomastercat")
            @Expose
            public String audiomastercat;
            @SerializedName("AudioSubCategory")
            @Expose
            public String audioSubCategory;
            @SerializedName("IsPlay")
            @Expose
            public String isPlay;
            @SerializedName("Bitrate")
            @Expose
            public String bitrate;
            public Detail() {
            }
            protected Detail(Parcel in) {
                ID = in.readString();
                name = in.readString();
                audioFile = in.readString();
                imageFile = in.readString();
                audioDuration = in.readString();
                audioDirection = in.readString();
                isPlay = in.readString();
                audiomastercat = in.readString();
                audioSubCategory = in.readString();
                bitrate = in.readString();
            }

            public static final Creator<MainAudioModel.ResponseData.Detail> CREATOR = new Creator<MainAudioModel.ResponseData.Detail>() {
                @Override
                public MainAudioModel.ResponseData.Detail createFromParcel(Parcel in) {
                    return new MainAudioModel.ResponseData.Detail(in);
                }

                @Override
                public MainAudioModel.ResponseData.Detail[] newArray(int size) {
                    return new MainAudioModel.ResponseData.Detail[size];
                }
            };

            public String getID() {
                return ID;
            }

            public void setID(String id) {
                this.ID = id;
            }

            public String getName() {
                return name;
            }

            public void setName(String name) {
                this.name = name;
            }

            public String getAudioFile() {
                return audioFile;
            }

            public void setAudioFile(String audioFile) {
                this.audioFile = audioFile;
            }

            public String getImageFile() {
                return imageFile;
            }

            public void setImageFile(String imageFile) {
                this.imageFile = imageFile;
            }

            public String getAudioDuration() {
                return audioDuration;
            }

            public void setAudioDuration(String audioDuration) {
                this.audioDuration = audioDuration;
            }

            public String getAudioDirection() {
                return audioDirection;
            }

            public void setAudioDirection(String audioDirection) {
                this.audioDirection = audioDirection;
            }

            public String getAudiomastercat() {
                return audiomastercat;
            }

            public void setAudiomastercat(String audiomastercat) {
                this.audiomastercat = audiomastercat;
            }

            public String getAudioSubCategory() {
                return audioSubCategory;
            }

            public void setAudioSubCategory(String audioSubCategory) {
                this.audioSubCategory = audioSubCategory;
            }

            public String getIsPlay() {
                return isPlay;
            }

            public void setIsPlay(String isPlay) {
                this.isPlay = isPlay;
            }

            public String getBitrate() {
                return bitrate;
            }

            public void setBitrate(String bitrate) {
                this.bitrate = bitrate;
            }

            @Override
            public int describeContents() {
                return 0;
            }

            @Override
            public void writeToParcel(Parcel parcel, int i) {
                parcel.writeString(ID);
                parcel.writeString(name);
                parcel.writeString(audioFile);
                parcel.writeString(imageFile);
                parcel.writeString(audioDuration);
                parcel.writeString(audioDirection);
                parcel.writeString(isPlay);
                parcel.writeString(audiomastercat);
                parcel.writeString(audioSubCategory);
                parcel.writeString(bitrate);
            }
        }
    }
}

File: app/src/main/java/com/brainwellnessspa/dashboardModule/models/ViewAllAudioListModel.kt
package com.brainwellnessspa.dashboardModule.models

import android.os.Parcel
import android.os.Parcelable
import android.os.Parcelable.Creator
import com.google.gson.annotations.Expose
import com.google.gson.annotations.SerializedName
import java.util.*

class ViewAllAudioListModel {
    @SerializedName("ResponseData")
    @Expose
    var responseData: ResponseData? = null

    @SerializedName("ResponseCode")
    @Expose
    var responseCode: String? = null

    @SerializedName("ResponseMessage")
    @Expose
    var responseMessage: String? = null

    @SerializedName("ResponseStatus")
    @Expose
    var responseStatus: String? = null

    class ResponseData() : Parcelable {
        @SerializedName("HomeAudioID")
        @Expose
        var homeAudioID: String? = null

        @SerializedName("View")
        @Expose
        var view: String? = null

        @SerializedName("UserId")
        @Expose
        var coUserId: String? = null

        @SerializedName("Details")
        @Expose
        var details: ArrayList<Detail>? = null

        constructor(parcel: Parcel) : this() {
            homeAudioID = parcel.readString()
            view = parcel.readString()
            coUserId = parcel.readString()
        }

        override fun describeContents(): Int {
            return 0
        }

        override fun writeToParcel(parcel: Parcel, i: Int) {
            parcel.writeString(homeAudioID)
            parcel.writeString(view)
            parcel.writeString(coUserId)
        }

        open class Detail : Parcelable {
            @SerializedName("ID")
            @Expose
            var iD: String? = null

            @SerializedName("Name")
            @Expose
            var name: String? = null

            @SerializedName("AudioFile")
            @Expose
            var audioFile: String? = null

            @SerializedName("ImageFile")
            @Expose
            var imageFile: String? = null

            @SerializedName("AudioDuration")
            @Expose
            var audioDuration: String? = null

            @SerializedName("AudioDirection")
            @Expose
            var audioDirection: String? = null

            @SerializedName("Audiomastercat")
            @Expose
            var audiomastercat: String? = null

            @SerializedName("AudioSubCategory")
            @Expose
            var audioSubCategory: String? = null

            @SerializedName("IsPlay")
            @Expose
            var isPlay: String? = null

            @SerializedName("Bitrate")
            @Expose
            var bitrate: String? = null

            constructor()
            protected constructor(`in`: Parcel) {
                iD = `in`.readString()
                name = `in`.readString()
                audioFile = `in`.readString()
                imageFile = `in`.readString()
                audioDuration = `in`.readString()
                audioDirection = `in`.readString()
                isPlay = `in`.readString()
                audiomastercat = `in`.readString()
                audioSubCategory = `in`.readString()
                bitrate = `in`.readString()
            }

            override fun describeContents(): Int {
                return 0
            }

            override fun writeToParcel(parcel: Parcel, i: Int) {
                parcel.writeString(iD)
                parcel.writeString(name)
                parcel.writeString(audioFile)
                parcel.writeString(imageFile)
                parcel.writeString(audioDuration)
                parcel.writeString(audioDirection)
                parcel.writeString(isPlay)
                parcel.writeString(audiomastercat)
                parcel.writeString(audioSubCategory)
                parcel.writeString(bitrate)
            }

            companion object {
                @JvmField val CREATOR: Creator<MainAudioModel.ResponseData.Detail?> = object : Creator<MainAudioModel.ResponseData.Detail?> {
                    override fun createFromParcel(`in`: Parcel): MainAudioModel.ResponseData.Detail {
                        return MainAudioModel.ResponseData.Detail(`in`)
                    }

                    override fun newArray(size: Int): Array<MainAudioModel.ResponseData.Detail?> {
                        return arrayOfNulls(size)
                    }
                }
            }
        }

        companion object {
            @JvmField val CREATOR: Creator<MainAudioModel.ResponseData?> = object : Creator<MainAudioModel.ResponseData?> {
                override fun createFromParcel(`in`: Parcel): MainAudioModel.ResponseData {
                    return MainAudioModel.ResponseData(`in`)
                }

                override fun newArray(size: Int): Array<MainAudioModel.ResponseData?> {
                    return arrayOfNulls(size)
                }
            }
        }
    }
}